scripts/lateralus.js

```diff
@@ -1,260 +1,239 @@
+  Proxy.prototype = parent.prototype;
+  child.prototype = new Proxy();
+  return child;
+};
+
+/**
+ * Create a `{@link Lateralus}` application instance.
+ *
+ *     var App = Lateralus.beget(function () {
+ *       Lateralus.apply(this, arguments);
+ *     });
+ * @static
+ * @method Lateralus.beget
+ * @param {Function} child
+ * @param {Object} [config]
+ * @param {LateralusModel} [config.Model] A `{@link Lateralus.Model}` subclass
+ * constructor to use for `{@link Lateralus.model}` instead of a standard
+ * `{@link Lateralus.Model}`.
+ * @return {Function} The created `{@link Lateralus}`
+ * subclass.
+ */
+Lateralus.beget = function (child, config) {
+  var lateralusConfig = config || {};
+
+  child.displayName = child.name || 'begetConstructor';
+  var begottenConstructor = Lateralus.inherit(child, Lateralus);
+  begottenConstructor.prototype.config = _.clone(lateralusConfig);
+
+  return begottenConstructor;
+};
+
+_.extend(fn, mixins);
+
+_.each([
 
   /**
-   * Relay `{{#crossLink "Lateralus.mixins/provide:property"}}{{/crossLink}}`d
-   * handlers to another `{{#crossLink "Lateralus"}}{{/crossLink}}` instance.
-   * This is the `{{#crossLink
-   * "Lateralus.mixins/provide:property"}}{{/crossLink}}` analog to
-   * `{{#crossLink "Lateralus.mixins/amplify"}}{{/crossLink}}`.
-   * @method shareWith
-   * @param {Lateralus} receiver The `{{#crossLink "Lateralus"}}{{/crossLink}}`
-   * instance to share `{{#crossLink
-   * "Lateralus.mixins/provide:property"}}{{/crossLink}}`d handlers with.
-   * @param {string} providerName The name of the `{{#crossLink
-   * "Lateralus.mixins/provide:property"}}{{/crossLink}}`er.
+   * Cross-browser friendly wrapper for `console.log`.
+   * @method Lateralus#log
+   * @param {...any} Any parameters to pass along to `console.log`.
    */
-  fn.shareWith = function (receiver, providerName) {
-    this.amplify(receiver, mixins.PROVIDE_PREFIX + providerName);
-  };
+  'log',
 
   /**
-   * Remove this `{{#crossLink "Lateralus"}}{{/crossLink}}` app from memory.
-   * @method dispose
+   * Cross-browser friendly wrapper for `console.warn`.
+   * @method Lateralus#warn
+   * @param {...any} Any parameters to pass along to `console.warn`.
    */
-  fn.dispose = function () {
-    _(this).lateralusDispose(_.bind(function () {
-      if (this.components) {
-        _.invoke(this.components, 'dispose');
-      }
-    }, this));
-  };
-  fn.spiralOut = fn.dispose;
+  'warn',
 
   /**
-   * Do not override this method, it is used internally.
-   * @method toString
-   * @return {string} This is `"lateralus"`.
-   * @final
+   * Cross-browser friendly wrapper for `console.error`.
+   * @method Lateralus#error
+   * @param {...any} Any parameters to pass along to `console.error`.
    */
-  fn.toString = function () {
-    return 'lateralus';
-  };
+  'error'
 
-  Lateralus.Component = Component;
-  Lateralus.Model = LateralusModel;
-  Lateralus.Router = LateralusRouter;
+], function (consoleMethodName) {
```

Comment:
You can preserve line numbers for easier debugging by using `bind()`. Here's a little  wrapper I wrote a while ago that does it:

```
var bind = Function.prototype.bind, // Console methods don't inherit from Function.prototype in IE9
	logger,
	wrapper;

// Polyfill while we're at it
if (typeof window.console === 'undefined') {
	window.console[consoleMethodName] = function() {};
}

logger = window.console; 

fn[consoleMethodName] = bind.call(logger[consoleMethodName], logger);

```